src/libraries/System.Text.RegularExpressions/tests/UnitTests/RegexPrefixAnalyzerTests.cs

```diff
@@ -70,6 +70,69 @@ public void FindFirstCharClass_StressDeep()
+        [InlineData("(ab+c|bcd+){3,5}", new[] { "ab", "bcd" }, false)]
+        [InlineData("abc|def", new[] { "abc", "def" }, false)]
+        [InlineData("ab{4}c|def{5}|g{2,4}h", new[] { "abbbbc", "defffff", "gg" }, false)]
+        [InlineData("abc|def|(ghi|jklm)", new[] { "abc", "def", "ghi", "jklm" }, false)]
+        [InlineData("abc[def]ghi", new[] { "abcdghi", "abceghi", "abcfghi" }, false)]
+        [InlineData("abc[def]ghi|[jkl]m", new[] { "abcdghi", "abceghi", "abcfghi", "jm", "km", "lm" }, false)]
+        [InlineData("agggtaaa|tttaccct", new[] { "agggtaaa", "tttaccct" }, false)]
+        [InlineData("[cgt]gggtaaa|tttaccc[acg]", new[] { "cgggtaaa", "ggggtaaa", "tgggtaaa", "tttaccca", "tttacccc", "tttacccg" }, false)]
+        [InlineData("a[act]ggtaaa|tttacc[agt]t", new[] { "aaggtaaa", "acggtaaa", "atggtaaa", "tttaccat", "tttaccgt", "tttacctt" }, false)]
+        [InlineData("ag[act]gtaaa|tttac[agt]ct", new[] { "agagtaaa", "agcgtaaa", "agtgtaaa", "tttacact", "tttacgct", "tttactct" }, false)]
+        [InlineData("agg[act]taaa|ttta[agt]cct", new[] { "aggataaa", "aggctaaa", "aggttaaa", "tttaacct", "tttagcct", "tttatcct" }, false)]
+        [InlineData("\b(abc|def)\b", new[] { "abc", "def" }, false)]
+        [InlineData("^(abc|def)$", new[] { "abc", "def" }, false)]
+        [InlineData("abcdefg|h", null, false)]
+        [InlineData("abc[def]ghi|[jkl]", null, false)]
+        [InlineData("[12][45][789]", new[] { "147", "148", "149", "157", "158", "159", "247", "248", "249", "257", "258", "259" }, false)]
+        [InlineData("[12]a[45]b[789]c", new[] { "1a4b7c", "1a4b8c", "1a4b9c", "1a5b7c", "1a5b8c", "1a5b9c", "2a4b7c", "2a4b8c", "2a4b9c", "2a5b7c", "2a5b8c", "2a5b9c" }, false)]
+        // case-insensitive
+        [InlineData("[Aa][Bb][Cc]", new[] { "abc" }, true)]
+        [InlineData("[Aa][Bbc][Cc]", null, true)]
+        [InlineData(":[Aa]![Bb]@", new[] { ":a!b@" }, true)]
+        [InlineData("(?i)abc", new[] { "abc" }, true)]
+        [InlineData("(?i)(abc+|bcd+)", new[] { "abc", "bcd" }, true)]
+        [InlineData("(?i)(ab+c|bcd+)", new[] { "ab", "bcd" }, true)]
+        [InlineData("(?i)(ab+c|bcd+)*", null, true)]
+        [InlineData("(?i)(ab+c|bcd+)+", new[] { "ab", "bcd" }, true)]
+        [InlineData("(?i)(ab+c|bcd+){3,5}", new[] { "ab", "bcd" }, true)]
+        [InlineData("(?i)abc|def", new[] { "abc", "def" }, true)]
+        [InlineData("(?i)ab{4}c|def{5}|g{2,4}h", new[] { "abbbbc", "defffff", "gg" }, true)]
+        [InlineData("(?i)(((?>abc)|(?>def)))", new[] { "abc", "def" }, true)]
+        [InlineData("(?i)(abc|def|(ghi|jklm))", null, true)]
+        [InlineData("(?i)(abc|def|(ghi|jlmn))", new[] { "abc", "def", "ghi", "jlmn" }, true)]
+        [InlineData("abc", null, true)]
+        [InlineData("abc|def", null, true)]
+        [InlineData("abc|def|(ghi|jklm)", null, true)]
+        [InlineData("://[Aa][Bb]|[Cc]@!", new[] { "://ab", "c@!" }, true)]
```

Comment:
Note to self: This is a case where `SearchValues<string>` could take `"://ab", "c@!"` and expand it into `"://ab", "c@!", "C@!"` and switch to case-sensitive Teddy prefix searching for a small speed boost.